build.ps1

```diff
@@ -9,36 +9,60 @@ param
 	[string]$configuration = "Release",	 
 	
     # target to build. 
-	#Options are: RebuildAll: Clean, Build all and Generate Nuget Packages. BuildAll: Build all csproj. GeneratePackages: Build all csproj and generate nuget packages.
-	[ValidateSet('Rebuildall', 'BuildAll', 'CleanAll', 'GeneratePackages', 'BuildTests')]
+	#Options are:
+	#   RebuildAll: Clean, Build all .csproj and Generate Nuget Packages. This is the default option.
+	#   BuildAll: Build all .csproj and Generate Nuget Packages.
+	#   GeneratePackages: Build all product .csprojs and generate nuget packages.
+	#   BuildTests: Builds all test .csprojs.
+	#   BuildProduct: Builds all product .csprojs.
+	[ValidateSet('Rebuildall', 'BuildAll', 'GeneratePackages', 'BuildTests', 'BuildProduct')]
 	[string]$target = "RebuildAll",
 
 	# msbuild verbosity level.
 	[ValidateSet('quiet','minimal', 'normal', 'detailed', 'diagnostic')]
 	[string]$verbosity = 'minimal'
 )
 
+
+
 # Check msbuild exists
-$msbuildRelativePath = "MSBuild\14.0\bin\MSBuild.exe"
-$msbuildAltPath = "Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\MSBuild.exe";
+$msbuildPath = "MSBuild\14.0\bin\MSBuild.exe"
 
 if (Test-Path "env:\ProgramFiles(x86)") 
 {
-    $msbuildPath = join-path ${env:ProgramFiles(x86)} $msbuildAltPath
-}
-elseif (Test-Path "env:\ProgramFiles(x86)") 
-{
-    $msbuildPath = join-path ${env:ProgramFiles(x86)} $msbuildRelativePath
+	$progFilesPath =  ${env:ProgramFiles(x86)}
 }
 elseif (Test-Path "env:\ProgramFiles") 
 {
-	$msbuildPath = join-path ${env:ProgramFiles(x86)} $msbuildRelativePath
+	$progFilesPath =  ${env:ProgramFiles(x86)}
```

Comment:
This should probably be 
`$progFilesPath =  ${env:ProgramFiles}`
based on the condition being tested.